src/xdocs/config_imports.xml

```diff
@@ -735,21 +735,51 @@ import android.*;
         </source>
 
         <p>
-          In the example below, all classes beginning with an I in the package
-          java.awt are allowed. In the package java.io only the classes File
-          and InputStream are allowed.
+            In the example below access to package
+            <code>com.puppycrawl.tools.checkstyle.checks</code> and its subpackages is
+            allowed from anywhere in <code>com.puppycrawl.tools.checkstyle</code> except
+            from the <code>filters</code> subpackage where access to all
+            <code>check</code>'s subpackages is disallowed. Two <code>java.lang.ref</code>
+            classes are allowed by virtue of one regular expression instead of listing
+            them in two separate allow rules (as it is done with the <code>Files</code>
+            and <code>ClassPath</code> classes).
         </p>
 
         <source>
 &lt;import-control pkg=&quot;com.puppycrawl.tools.checkstyle&quot;&gt;
-    &lt;allow class=&quot;java\.awt\.I.*&quot; regex=&quot;true&quot;/&gt;
-    &lt;allow class=&quot;java\.io\.(File|InputStream)&quot; local-only=&quot;true&quot;
-        regex=&quot;true&quot;/&gt;
+    &lt;allow pkg=&quot;com.puppycrawl.tools.checkstyle.api&quot;/&gt;
+    &lt;allow pkg=&quot;com.puppycrawl.tools.checkstyle.checks&quot;/&gt;
+    &lt;allow class=&quot;com.google.common.io.Files&quot;/&gt;
+    &lt;allow class=&quot;com.google.common.reflect.ClassPath&quot;/&gt;
+    &lt;subpackage name=&quot;filters&quot;&gt;
+        &lt;allow class=&quot;java\.lang\.ref\.(Weak|Soft)Reference&quot;
+            regex=&quot;true&quot;/&gt;
+        &lt;disallow pkg=&quot;com\.puppycrawl\.tools\.checkstyle\.checks\.[^.]+&quot;
+            regex=&quot;true&quot;/&gt;
+        &lt;disallow pkg=&quot;com.puppycrawl.tools.checkstyle.ant&quot;/&gt;
+        &lt;disallow pkg=&quot;com.puppycrawl.tools.checkstyle.doclets&quot;/&gt;
+        &lt;disallow pkg=&quot;com.puppycrawl.tools.checkstyle.gui&quot;/&gt;
+    &lt;/subpackage&gt;
 &lt;/import-control&gt;
         </source>
 
         <p>
-          For an example import control file, look at the file called <a
+          In the next example regular expressions are used to enforce a layering rule: In
+          all modules it forbids access to all <code>service</code> packages from all
+          <code>dao</code> packages. The root package is also a regular expression that is
+          used to handle old and new domain name with the same rules.
+        </p>
```

Comment:
thanks for example, but lets make it more general is understandable for all.
Lets update it a bit .... and forbid usage of dao/service packages in ui packages and vise-versa.

```
<import-control pkg="(de.olddomain|de.newdomain)\..*" regex="true">
    <subpackage pkg="[^.]+\.dao" regex="true">
        <disallow pkg=".*\.ui" regex="true"/>
        <disallow pkg=".*\.(awt|swing).\.*" regex="true"/>
    </subpackage>
    <subpackage pkg="[^.]+\.ui" regex="true">
        <disallow pkg=".*\.(dao|service)" regex="true"/>
    </subpackage>
</import-control>
```